base/linalg/eigen.jl

```diff
@@ -27,6 +27,45 @@ function getindex(A::Union{Eigen,GeneralizedEigen}, d::Symbol)
     throw(KeyError(d))
 end
 
+eigsortby(λ::Real) = λ
+eigsortby(λ::Complex) = reim(λ)
+
+"""
+    sort!(F::Base.LinAlg.Eigen; kw...)
+
+Sort the eigenvectors and eigenvalues of an eigenfactorization `F` in place using the eigenvalues
+for comparison.  See also `sort!(::AbstractVector)`.
+
+# Examples
```

Comment:
Should the example perhaps be attached to the `sort` method instead? It is a fairly common pattern throughout the documentation to have a more extensive docstring for the non-inplace method, and just reference that there is an inplace equivalent.

Reply:
I think that makes sense.  But for `sort(::Vector)`, it actually references `sort!` as the extensive docstring.  Maybe this should be changed?

Reply:
Hmm okay, I guess we don't have an official policy, in any case it is unnecessary to repeat information, so one or the other should have a more extensive docstring. Perhaps you can simply add a short docstring for the non-inplace version instead then, and reference this. I guess most of the time you want to use `sort!` anyway.

Also, why the reference to `sort!(::AbstractVector)`? For docs about the keywords?